ls1intum/Artemis
# `Iris`: Add connector to Pyris

### Checklist
#### General

#### Server

### Motivation and Context



### Description
This PR replaces the IrisModels and the IrisModelService with the IrisConnectorService which connects the server of Artemis with an Iris implementation in Python called Pyris. Through Pyris, we can utilize Microsoft's Guidance library.

### Steps for Testing
Prerequesites:
- Programming Exercise with Iris enabled. To enable Iris for a programming exercise, it also needs to be enabled for the course and server-wide.
- Student with access to the course with the aforementioned programming exercise

1. Open the exercise and click on the Iris icon bottom right.
2. Accept the terms and conditions
3. Chat with Iris and assert that e.g., the answers are not cut off, off-topic answers are not answered, Iris has knowledge about the exercise and the course

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests








| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [IrisAuthorizationInterceptor.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.config.auth/IrisAuthorizationInterceptor.html) | 100% | ✅ |
| [IrisConnectorException.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris/IrisConnectorException.html) | 100% | ✅ |
| [IrisConnectorService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris/IrisConnectorService.html) | 91% | ✅ |
| [IrisModel.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris/IrisModel.html) | 100% | ✅ |
| [IrisMessageResponseDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris.dto/IrisMessageResponseDTO.html) | 100% | ✅ |
| [IrisRequestDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris.dto/IrisRequestDTO.html) | 100% | ✅ |
| [IrisMessageService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisMessageService.html) | 100% | ✅ |
| [IrisSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisSessionService.html) | 90% | ✅ |
| [IrisSettingsService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisSettingsService.html) | 88% | ✅ |
| [IrisChatSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.session/IrisChatSessionService.html) | 89% | ✅ |
| [IrisHestiaSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.session/IrisHestiaSessionService.html) | 90% | ✅ |
| [IrisMessageResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.iris/IrisMessageResource.html) | 100% | ✅ |
| [IrisSessionResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4991-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.iris/IrisSessionResource.html) | 100% | ✅ |
